lfucg-exactions/server/plats/views.py

```diff
@@ -1,3 +1,135 @@
 from django.shortcuts import render
+import csv
+from django.views.generic import View
+from django.http import HttpResponse
 
-# Create your views here.
+from .models import *
+from .serializers import *
+from accounts.models import *
+from .utils import calculate_lot_balance
+
+class PlatCSVExportView(View):
+     def get(self, request, *args, **kwargs):
+        headers = ['Subdivision',
+            'Developer Account',
+            'Total Acreage',
+            'Buildable Lots',
+            'Non-Buildable Lots',
+            'Plat Type',
+            'Expansion Area',
+            'Unit',
+            'Block',
+            'Section',
+            'Cabinet',
+            'Slide Number',
+            'Non-Sewer Exactions',
+            'Sewer Exactions',
+        ]
+
+        plat = request.GET.get('plat')
+
+        response = HttpResponse(content_type='text')
+        response['Content-Disposition'] = 'attachment; filename="plat_report.csv"'
+
+        plat_queryset = Plat.objects.filter(id=plat)[0]
+
+        writer = csv.writer(response)
+
+        plat_csv_data = [ plat_queryset.subdivision.name,
+            plat_queryset.account,
+            plat_queryset.total_acreage,
+            plat_queryset.buildable_lots,
+            plat_queryset.non_buildable_lots,
+            plat_queryset.plat_type,
+            plat_queryset.expansion_area,
+            plat_queryset.unit,
+            plat_queryset.block,
+            plat_queryset.section,
+            plat_queryset.cabinet,
+            plat_queryset.slide,
+            plat_queryset.non_sewer_due,
+            plat_queryset.sewer_due,
+        ]
+
+        # LOTS AND LOT DETAILS
+        lot_queryset = Lot.objects.filter(plat=plat)
+
+        if lot_queryset is not None:
+            headers.extend([
+                'Address',
+                'Lot Number',
+                'Parcel ID',
+                'Permit ID', 
+                'Latitude',
+                'Longitude',
+                'Roads',
+                'Parks',
+                'Storm Water',
+                'Open Spaces',
+                'Sewer Cap.',
+                'Sewer Trans.',
+                'Total Exactions',
+                'Sewer Exactions',
+                'Non-Sewer Exactions',
+                'Current  Exactions',
+                'Current Sewer Exactions',
+                'Current Non-Sewer Exactions',
+                'Payment Total - 1',
```

Comment:
Is this just to show multiple payments separately?  Setting an arbitrary limit on something like this is usually a bad idea.  I think it would be fine to show the payment sum in one column.  (I suppose this is okay IF this is something Phyllis's spreadsheet already does)